modules/openideal_idea/src/Form/OpenidealIdeaAddFiveStarsForm.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+namespace Drupal\openideal_idea\Form;
+
+use Drupal\Component\Utility\Html;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\field\Entity\FieldStorageConfig;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Configure OpenideaL Idea settings for this site.
+ */
+class OpenidealIdeaAddFiveStarsForm extends FormBase {
+
+  protected const ENTITY_TYPE = 'node';
+
+  protected const FIELD_TYPE = 'voting_api_field';
+
+  protected const BUNDLE = 'idea';
+
+  /**
+   * Entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityManager;
+
+  /**
+   * Entity Field Manager.
+   *
+   * @var \Drupal\Core\Entity\EntityFieldManager
+   */
+  protected $entityFieldManager;
+
+  /**
+   * {@inheritDoc}
+   */
+  public static function create(ContainerInterface $container) {
+    $instance = parent::create($container);
+    $instance->entityManager = $container->get('entity_type.manager');
+    $instance->entityFieldManager = $container->get('entity_field.manager');
+
+    return $instance;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'openideal_idea_openideal_idea_add_five_stars_widget';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $form['#id'] = Html::getClass($this->getFormId());
+    $field_prefix = $this->config('field_ui.settings')->get('field_prefix');
+    $form['#tree'] = TRUE;
+    $form['field'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Widget'),
+      'label' => [
+        '#required' => TRUE,
+        '#type' => 'textfield',
+        '#title' => $this->t('Label'),
+        '#default_value' => '',
+      ],
+      'weight' => [
+        '#title' => $this->t('Weight'),
+        '#description' => $this->t('The widget weight directly affect overall idea score'),
+        '#type' => 'textfield',
+        '#default_value' => 0,
+      ],
+      'field_name' => [
```

Comment:
Can we mention in the comment that this code was taken from `FieldStorageAddForm`?